### PR TITLE
Fix moment group indicators in text diagrams

### LINKED ISSUE #1467: Missing moment indicators in circuit diagram
This code:

```python
import cirq.experiments
print(cirq.experiments.generate_supremacy_circuit_google_v2_grid(
    n_rows=4,
    n_cols=5,
    cz_depth=20,
    seed=1
).to_text_diagram(transpose=True))
```

Produces this diagram:

```
  (0, 0) (0, 1) (0, 2) (0, 3) (0, 4) (1, 0) (1, 1) (1, 2) (1, 3) (1, 4) (2, 0) (2, 1) (2, 2) (2, 3) (2, 4) (3, 0) (3, 1) (3, 2) (3, 3) (3, 4)
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  H      H      H      H      H      H      H      H      H      H      H      H      H      H      H      H      H      H      H      H
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  @──────@      T      T      T      T      T      @──────@      T      @──────@      T      T      T      T      T      @──────@      T
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  X^0.5  Y^0.5  │      │      │      @──────┼──────┼──────┼──────┼──────@      X^0.5  │      │      │      │      │      X^0.5  X^0.5  │
  │      │      │      │      │      │      │      │      Y^0.5  @──────┼──────┼──────┼──────┼──────@      │      │      │      │      │
  │      │      │      │      │      │      │      @──────┼──────┼──────┼──────┼──────@      │      │      │      │      │      │      │
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  T      @──────@      │      │      Y^0.5  │      Y^0.5  @──────@      X^0.5  @──────@      │      X^0.5  │      │      T      @──────@
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  │      Y^0.5  @──────┼──────┼──────┼──────┼──────@      X^0.5  │      T      │      Y^0.5  @──────┼──────┼──────┼──────┼──────@      X^0.5
  @──────┼──────┼──────┼──────┼──────@      │      │      │      │      │      @──────┼──────┼──────┼──────┼──────@      │      │      │
  │      │      │      │      @──────┼──────┼──────┼──────┼──────@      │      │      │      │      T      │      │      │      │      │
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  X^0.5  T      @──────@      Y^0.5  @──────@      X^0.5  T      Y^0.5  │      Y^0.5  @──────@      │      @──────@      │      X^0.5  T
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  T      │      X^0.5  Y^0.5  T      Y^0.5  @──────┼──────┼──────┼──────┼──────@      X^0.5  │      │      X^0.5  X^0.5  │      T      │
  │      │      │      │      │      │      │      T      @──────┼──────┼──────┼──────┼──────@      │      │      │      │      │      │
  │      │      │      │      │      │      │      │      │      T      │      │      │      │      │      │      │      │      │      │
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  │      │      T      @──────@      T      @──────@      X^0.5  │      │      X^0.5  T      @──────@      T      @──────@      │      │
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
┌╴│      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │     ╶┐
│ │      @──────┼──────┼──────┼──────┼──────@      X^0.5  │      │      │      T      │      X^0.5  @──────┼──────┼──────┼──────┼──────@      │
│ │      │      │      @──────┼──────┼──────┼──────┼──────@      │      @──────┼──────┼──────┼──────┼──────@      X^0.5  │      │      │      │
└╴│      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │     ╶┘
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  │      │      │      │      X^0.5  │      │      │      │      │      │      │      @──────┼──────┼──────┼──────┼──────@      │      │
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  @──────@      │      X^0.5  T      │      X^0.5  @──────@      │      @──────@      X^0.5  T      X^0.5  Y^0.5  T      @──────@      Y^0.5
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  Y^0.5  X^0.5  │      T      │      @──────┼──────┼──────┼──────┼──────@      Y^0.5  │      │      │      T      │      X^0.5  X^0.5  T
  │      │      │      │      │      │      T      │      Y^0.5  @──────┼──────┼──────┼──────┼──────@      │      │      │      │      │
  │      │      │      │      │      │      │      @──────┼──────┼──────┼──────┼──────@      │      │      │      │      │      │      │
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  T      @──────@      │      │      Y^0.5  │      Y^0.5  @──────@      Y^0.5  @──────@      │      X^0.5  │      │      T      @──────@
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  │      Y^0.5  @──────┼──────┼──────┼──────┼──────@      Y^0.5  │      T      │      X^0.5  @──────┼──────┼──────┼──────┼──────@      Y^0.5
  @──────┼──────┼──────┼──────┼──────@      │      │      │      │      │      @──────┼──────┼──────┼──────┼──────@      │      │      │
  │      │      │      │      @──────┼──────┼──────┼──────┼──────@      │      │      │      │      T      │      │      │      │      │
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  Y^0.5  T      @──────@      Y^0.5  @──────@      Y^0.5  T      Y^0.5  │      X^0.5  @──────@      │      @──────@      │      X^0.5  T
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  T      │      Y^0.5  X^0.5  T      X^0.5  @──────┼──────┼──────┼──────┼──────@      Y^0.5  │      │      Y^0.5  Y^0.5  │      T      │
  │      │      │      │      │      │      │      T      @──────┼──────┼──────┼──────┼──────@      │      │      │      │      │      │
  │      │      │      │      │      │      │      │      │      T      │      │      │      │      │      │      │      │      │      │
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  │      │      T      @──────@      T      @──────@      X^0.5  │      │      X^0.5  T      @──────@      T      @──────@      │      │
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
┌╴│      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │     ╶┐
│ │      @──────┼──────┼──────┼──────┼──────@      Y^0.5  │      │      │      T      │      Y^0.5  @──────┼──────┼──────┼──────┼──────@      │
│ │      │      │      @──────┼──────┼──────┼──────┼──────@      │      @──────┼──────┼──────┼──────┼──────@      X^0.5  │      │      │      │
└╴│      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │     ╶┘
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  │      │      │      │      Y^0.5  │      │      │      │      │      │      │      @──────┼──────┼──────┼──────┼──────@      │      │
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  @──────@      │      X^0.5  T      │      X^0.5  @──────@      │      @──────@      X^0.5  T      Y^0.5  Y^0.5  T      @──────@      Y^0.5
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  X^0.5  X^0.5  │      T      │      @──────┼──────┼──────┼──────┼──────@      Y^0.5  │      │      │      T      │      Y^0.5  Y^0.5  T
  │      │      │      │      │      │      T      │      Y^0.5  @──────┼──────┼──────┼──────┼──────@      │      │      │      │      │
  │      │      │      │      │      │      │      @──────┼──────┼──────┼──────┼──────@      │      │      │      │      │      │      │
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  T      @──────@      │      │      Y^0.5  │      X^0.5  @──────@      Y^0.5  @──────@      │      Y^0.5  │      │      T      @──────@
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  │      Y^0.5  @──────┼──────┼──────┼──────┼──────@      X^0.5  │      T      │      X^0.5  @──────┼──────┼──────┼──────┼──────@      Y^0.5
  @──────┼──────┼──────┼──────┼──────@      │      │      │      │      │      @──────┼──────┼──────┼──────┼──────@      │      │      │
  │      │      │      │      @──────┼──────┼──────┼──────┼──────@      │      │      │      │      T      │      │      │      │      │
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  H      H      H      H      H      H      H      H      H      H      H      H      H      H      H      H      H      H      H      H
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
```

Look carefully at lines 7-9 of the diagram:

```
  X^0.5  Y^0.5  │      │      │      @──────┼──────┼──────┼──────┼──────@      X^0.5  │      │      │      │      │      X^0.5  X^0.5  │
  │      │      │      │      │      │      │      │      Y^0.5  @──────┼──────┼──────┼──────┼──────@      │      │      │      │      │
  │      │      │      │      │      │      │      @──────┼──────┼──────┼──────┼──────@      │      │      │      │      │      │      │
```

These three lines all have gates; they are not vertically separated by a layer with no gates. This indicates they came from the same moment. But there is no same-moment indicator on the left or right.

Either these lines should not be placed together as they are, or the moment indicator should have been included.

### PR DESCRIPTION
**Pull request by Ashalynd**
This fixes the issue, but inserts an additional line after newly detected group. As a result, I had to change expectation for a test where a similarly-looking (at least, to me) group also wasn't detected.

Fixes #1467

### CODE REVIEW

**Comment by Ashalynd** [comment]
I have used the circuit proposed by davemc84@, but the example mentioned in the issue seemed to work too:

```
>>> import cirq.experiments
>>> print(cirq.experiments.generate_supremacy_circuit_google_v2_grid(
...     n_rows=4,
...     n_cols=5,
...     cz_depth=20,
...     seed=1
... ).to_text_diagram(transpose=True))
  (0, 0) (0, 1) (0, 2) (0, 3) (0, 4) (1, 0) (1, 1) (1, 2) (1, 3) (1, 4) (2, 0) (2, 1) (2, 2) (2, 3) (2, 4) (3, 0) (3, 1) (3, 2) (3, 3) (3, 4)
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  H      H      H      H      H      H      H      H      H      H      H      H      H      H      H      H      H      H      H      H
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  @──────@      T      T      T      T      T      @──────@      T      @──────@      T      T      T      T      T      @──────@      T
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
┌╴│      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │     ╶┐
│ X^0.5  Y^0.5  │      │      │      │      │      │      Y^0.5  @──────┼──────┼──────┼──────┼──────@      │      │      X^0.5  X^0.5  │      │
│ │      │      │      │      │      @──────┼──────┼──────┼──────┼──────@      X^0.5  │      │      │      │      │      │      │      │      │
│ │      │      │      │      │      │      │      @──────┼──────┼──────┼──────┼──────@      │      │      │      │      │      │      │      │
└╴│      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │     ╶┘
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  T      @──────@      │      │      Y^0.5  │      Y^0.5  @──────@      X^0.5  @──────@      │      X^0.5  │      │      T      @──────@
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
┌╴│      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │     ╶┐
│ @──────┼──────┼──────┼──────┼──────@      │      │      X^0.5  │      T      │      Y^0.5  @──────┼──────┼──────┼──────┼──────@      X^0.5  │
│ │      Y^0.5  @──────┼──────┼──────┼──────┼──────@      │      │      │      @──────┼──────┼──────┼──────┼──────@      │      │      │      │
│ │      │      │      │      @──────┼──────┼──────┼──────┼──────@      │      │      │      │      T      │      │      │      │      │      │
└╴│      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │     ╶┘
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  X^0.5  T      @──────@      Y^0.5  @──────@      X^0.5  T      Y^0.5  │      Y^0.5  @──────@      │      @──────@      │      X^0.5  T
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
┌╴│      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │     ╶┐
│ T      │      X^0.5  Y^0.5  T      Y^0.5  │      T      @──────┼──────┼──────┼──────┼──────@      │      X^0.5  X^0.5  │      T      │      │
│ │      │      │      │      │      │      @──────┼──────┼──────┼──────┼──────@      X^0.5  │      │      │      │      │      │      │      │
│ │      │      │      │      │      │      │      │      │      T      │      │      │      │      │      │      │      │      │      │      │
└╴│      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │     ╶┘
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  │      │      T      @──────@      T      @──────@      X^0.5  │      │      X^0.5  T      @──────@      T      @──────@      │      │
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
┌╴│      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │     ╶┐
│ │      │      │      @──────┼──────┼──────┼──────┼──────@      │      │      T      │      X^0.5  @──────┼──────┼──────┼──────┼──────@      │
│ │      @──────┼──────┼──────┼──────┼──────@      X^0.5  │      │      @──────┼──────┼──────┼──────┼──────@      X^0.5  │      │      │      │
│ │      │      │      │      X^0.5  │      │      │      │      │      │      │      @──────┼──────┼──────┼──────┼──────@      │      │      │
└╴│      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │     ╶┘
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  @──────@      │      X^0.5  T      │      X^0.5  @──────@      │      @──────@      X^0.5  T      X^0.5  Y^0.5  T      @──────@      Y^0.5
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
┌╴│      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │     ╶┐
│ Y^0.5  X^0.5  │      T      │      │      T      │      Y^0.5  @──────┼──────┼──────┼──────┼──────@      T      │      X^0.5  X^0.5  T      │
│ │      │      │      │      │      @──────┼──────┼──────┼──────┼──────@      Y^0.5  │      │      │      │      │      │      │      │      │
│ │      │      │      │      │      │      │      @──────┼──────┼──────┼──────┼──────@      │      │      │      │      │      │      │      │
└╴│      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │     ╶┘
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  T      @──────@      │      │      Y^0.5  │      Y^0.5  @──────@      Y^0.5  @──────@      │      X^0.5  │      │      T      @──────@
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
┌╴│      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │     ╶┐
│ @──────┼──────┼──────┼──────┼──────@      │      │      Y^0.5  │      T      │      X^0.5  @──────┼──────┼──────┼──────┼──────@      Y^0.5  │
│ │      Y^0.5  @──────┼──────┼──────┼──────┼──────@      │      │      │      @──────┼──────┼──────┼──────┼──────@      │      │      │      │
│ │      │      │      │      @──────┼──────┼──────┼──────┼──────@      │      │      │      │      T      │      │      │      │      │      │
└╴│      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │     ╶┘
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  Y^0.5  T      @──────@      Y^0.5  @──────@      Y^0.5  T      Y^0.5  │      X^0.5  @──────@      │      @──────@      │      X^0.5  T
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
┌╴│      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │     ╶┐
│ T      │      Y^0.5  X^0.5  T      X^0.5  │      T      @──────┼──────┼──────┼──────┼──────@      │      Y^0.5  Y^0.5  │      T      │      │
│ │      │      │      │      │      │      @──────┼──────┼──────┼──────┼──────@      Y^0.5  │      │      │      │      │      │      │      │
│ │      │      │      │      │      │      │      │      │      T      │      │      │      │      │      │      │      │      │      │      │
└╴│      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │     ╶┘
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  │      │      T      @──────@      T      @──────@      X^0.5  │      │      X^0.5  T      @──────@      T      @──────@      │      │
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
┌╴│      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │     ╶┐
│ │      │      │      @──────┼──────┼──────┼──────┼──────@      │      │      T      │      Y^0.5  @──────┼──────┼──────┼──────┼──────@      │
│ │      @──────┼──────┼──────┼──────┼──────@      Y^0.5  │      │      @──────┼──────┼──────┼──────┼──────@      X^0.5  │      │      │      │
│ │      │      │      │      Y^0.5  │      │      │      │      │      │      │      @──────┼──────┼──────┼──────┼──────@      │      │      │
└╴│      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │     ╶┘
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  @──────@      │      X^0.5  T      │      X^0.5  @──────@      │      @──────@      X^0.5  T      Y^0.5  Y^0.5  T      @──────@      Y^0.5
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
┌╴│      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │     ╶┐
│ X^0.5  X^0.5  │      T      │      │      T      │      Y^0.5  @──────┼──────┼──────┼──────┼──────@      T      │      Y^0.5  Y^0.5  T      │
│ │      │      │      │      │      @──────┼──────┼──────┼──────┼──────@      Y^0.5  │      │      │      │      │      │      │      │      │
│ │      │      │      │      │      │      │      @──────┼──────┼──────┼──────┼──────@      │      │      │      │      │      │      │      │
└╴│      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │     ╶┘
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  T      @──────@      │      │      Y^0.5  │      X^0.5  @──────@      Y^0.5  @──────@      │      Y^0.5  │      │      T      @──────@
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
┌╴│      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │     ╶┐
│ @──────┼──────┼──────┼──────┼──────@      │      │      X^0.5  │      T      │      X^0.5  @──────┼──────┼──────┼──────┼──────@      Y^0.5  │
│ │      Y^0.5  @──────┼──────┼──────┼──────┼──────@      │      │      │      @──────┼──────┼──────┼──────┼──────@      │      │      │      │
│ │      │      │      │      @──────┼──────┼──────┼──────┼──────@      │      │      │      │      T      │      │      │      │      │      │
└╴│      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │     ╶┘
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │
  H      H      H      H      H      H      H      H      H      H      H      H      H      H      H      H      H      H      H      H
  │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │      │

```

File: cirq/circuits/circuit_test.py [hunk 1]
```diff
@@ -2966,3 +2970,35 @@ def test_pow_valid_only_for_minus_1():
...
+            ││     │││     │││     ││
+(0, 4): ────@┼─────@┼┼─────@┼┼─────┼┼────
+             │      ││      ││     ││
+(0, 5): ─────@──────@┼──────@┼─────@┼────
+                     │       │      │
+(0, 6): ─────────────@───────@──────@────
+
+(0, 7): ────H──────H─────────────────────
+           └──┘   └───┘   └───┘   └──┘
+""", use_unicode_characters=True)
```
Comment (vtomole) [inline cirq/circuits/circuit_test.py]: Need a newline

Comment (Ashalynd) [reply cirq/circuits/circuit_test.py]: Done

**Comment by Strilanc** [comment]
Remember to delete branches when merging.

**Pull request merged**